mp2i/cogs/sanctions.py

```diff
@@ -35,74 +61,111 @@ async def warn(self, ctx, member: discord.Member, dm: str, visible: str, *,
-                date=datetime.now(),
-                type="warn",
-                reason=reason,
-            )
+        self.__register_sanction_in_database(
+            staff=ctx.author.id,
+            member=member.id,
+            guild=ctx.guild.id,
+            date=datetime.now(),
+            type_="warn",
+            duration=None,
+            reason=reason,
         )
-        if dm == "oui":
-            if reason:
+        send_dm = dm == "oui"
+        message_sent = False
+        if send_dm:
+            # Au cas où l'utilisateur visé a fermé ses messages privés.
+            try:
                 await member.send(
                     "Vous avez reçu un avertissement pour la raison suivante: \n"
                     f">>> {reason}"
                 )
-            else:
-                await member.send("Vous avez reçu un avertissement.")
+                message_sent = True
+            except discord.Forbidden:
+                message_sent = False
 
         embed = discord.Embed(
-            title=f"{member.mention} a reçu un avertissement",
+            title=f"{member.name} a reçu un avertissement",
             colour=0xFF00FF,
             timestamp=datetime.now(),
         )
-        if reason:
-            embed.add_field(name="Raison", value=reason)
-        # If ephemeral is True, the message will only be visible to the author
-        await ctx.send(embed=embed, ephemeral=visible != "oui")
+        embed.add_field(name="Utilisateur", value=member.mention)
+        embed.add_field(name="Staff", value=ctx.author.mention)
+        embed.add_field(name="Raison", value=reason, inline=False)
+        if send_dm and message_sent:
+            embed.add_field(
+                name="Message privé", value="L'utilisateur a été averti.", inline=False
+            )
+        elif send_dm:
+            embed.add_field(
+                name="Message privé",
+                value=r"/!\ Aucun message n'a pu être envoyé à l'utilisateur.",
+                inline=False,
+            )
+
+        await ctx.send(embed=embed, ephemeral=True)
+
+        guild = GuildWrapper(ctx.guild)
+        if not guild.sanctions_log_channel:
+            return
+        await guild.sanctions_log_channel.send(embed=embed)
 
-    @hybrid_command(name="warnlist")
+    @hybrid_command(name="sanctionlist")
     @guild_only()
     @has_any_role("Modérateur", "Administrateur")
-    async def warnlist(self, ctx, member: Optional[discord.Member]) -> None:
+    @choices(
+        type_=[
+            Choice(name="Tout type", value="*"),
+            Choice(name="Avertissement", value="warn"),
+            Choice(name="Bâillonnage", value="to"),
+            Choice(name="Bannissement", value="ban"),
+            Choice(name="Débâillonnage", value="unto"),
+            Choice(name="Débannissement", value="unban"),
+        ]
+    )
+    async def sanction_list(
+        self, ctx, type_: str, member: Optional[discord.Member]
+    ) -> None:
```

Comment:
```suggestion
    @choices(
        type=[
            Choice(name="Tout type", value="*"),
            Choice(name="Avertissement", value="warn"),
            Choice(name="Exclusion temporaire", value="to"),
            Choice(name="Bannissement", value="ban"),
            Choice(name="Réintégration", value="unto"),
            Choice(name="Débannissement", value="unban"),
        ]
    )
    async def sanction_list(
        self, ctx, type: str, member: Optional[discord.Member]
    ) -> None:
```
Je reprends les mots de Discord parce que bâillonnage c'est très bizarre quand même :joy:

Reply:
Pour le type_, je ne comprends pas trop pourquoi tu as mit le _ à la fin, parce que on le voit aussi sur discord quand on fait la commande / et c'est un peu moche.
Mais du coup faut aussi remplacer toutes les autres variables qui suivent